src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,97 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int capacity;
+    private float loadFactor;
+    private Node<K, V>[] buckets;
+
+    public MyHashMap() {
+        this(DEFAULT_CAPACITY, LOAD_FACTOR);
+    }
+
+    public MyHashMap(int capacity, float loadFactor) {
+        this.capacity = capacity;
+        this.loadFactor = loadFactor;
+        this.buckets = new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int bucketIndex = getBucketIndex(key);
+        Node<K, V> node = buckets[bucketIndex];
+        if (needsResize()) {
+            resize();
+        }
```

Comment:
it is better to get the desired basket index after resizing, because the index may change and put you in the wrong basket